src/main/java/org/eclipse/uprotocol/cloudevent/factory/UCloudEvent.java

```diff
@@ -467,24 +469,14 @@ static CloudEvent fromMessageParts(UUri source, UAttributes attributes, UPayload
     static UPayloadFormat getUPayloadFormatFromContentType(String contentType){
         if(contentType == null)
             return UPayloadFormat.UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY;
-
-        switch (contentType){
-            case "application/json":
-                return UPayloadFormat.UPAYLOAD_FORMAT_JSON;
-            case "application/octet-stream":
-                return UPayloadFormat.UPAYLOAD_FORMAT_RAW;
-            case "text/plain":
-                return UPayloadFormat.UPAYLOAD_FORMAT_TEXT;
-            case "application/x-someip":
-                return UPayloadFormat.UPAYLOAD_FORMAT_SOMEIP;
-            case "application/x-someip_tlv":
-                return UPayloadFormat.UPAYLOAD_FORMAT_SOMEIP_TLV;
-            case "application/protobuf":
-                return UPayloadFormat.UPAYLOAD_FORMAT_PROTOBUF;
-            default:
-                return UPayloadFormat.UPAYLOAD_FORMAT_PROTOBUF_WRAPPED_IN_ANY;
-
+        
+        for (EnumValueDescriptor v : UPayloadFormat.getDescriptor().getValues()) {
```

Comment:
it is a lot harder to understand what the code is doing, can you add some JavaDoc to explain?

Reply:
Added better English :-)